tutorials/tts/FastPitch_Speaker_Interpolation.ipynb

```diff
@@ -0,0 +1,436 @@
+   "source": [
+    "Now we will define a function which will take as input the spectrogram generator model, vocoder model, text to be converted to speech and speaker id. This function will return spectrogram and audio of the generated speech."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9b6e10fd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def infer(spec_gen_model, vocoder_model, str_input, speaker=None):\n",
+    "    \"\"\"\n",
+    "    Synthesizes spectrogram and audio from a text string given a spectrogram synthesis and vocoder model.\n",
+    "    \n",
+    "    Args:\n",
+    "        spec_gen_model: Spectrogram generator model (FastPitch in our case)\n",
+    "        vocoder_model: Vocoder model (HiFiGAN in our case)\n",
+    "        str_input: Text input for the synthesis\n",
+    "        speaker: Speaker ID\n",
+    "    \n",
+    "    Returns:\n",
+    "        spectrogram and waveform of the synthesized audio.\n",
+    "    \"\"\"\n",
+    "    spec_gen_model.eval()\n",
+    "    with torch.no_grad():\n",
+    "        parsed = spec_gen_model.parse(str_input)\n",
+    "        if speaker is not None:\n",
+    "            speaker = torch.tensor([speaker]).long().to(device=spec_gen_model.device)\n",
+    "        spectrogram = spec_gen_model.generate_spectrogram(tokens=parsed, speaker=speaker)\n",
+    "        audio = vocoder_model.convert_spectrogram_to_audio(spec=spectrogram)\n",
+    "        \n",
+    "    if spectrogram is not None:\n",
+    "        if isinstance(spectrogram, torch.Tensor):\n",
+    "            spectrogram = spectrogram.to('cpu').numpy()\n",
+    "        if len(spectrogram.shape) == 3:\n",
+    "            spectrogram = spectrogram[0]\n",
+    "    if isinstance(audio, torch.Tensor):\n",
+    "        audio = audio.to('cpu').numpy()\n",
+    "    return spectrogram, audio"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89b17e26",
+   "metadata": {},
+   "source": [
+    "Now let's generate some speech."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a8d966d6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Choose any original speakers (speaker1 and speaker2)\n",
+    "# Choose new_speaker ID such that it is not already assigned to an existing speaker\n",
+    "speaker1, speaker2, new_speaker = 6097, 92, 1000\n",
+    "\n",
+    "# Choose weight1 corresponding to speaker1 and weight2 corresponding to speaker2\n",
+    "# You can change this and play around with the weights\n",
+    "weight1, weight2 = 0.4, 0.6\n",
+    "\n",
+    "# Generate new speaker embedding using speaker interpolation\n",
+    "new_speaker_emb = get_interpolated_speaker_emb(spec_gen_model, speaker1, speaker2, weight1, weight2)\n",
+    "# Assign the new speaker embedding to the new_speaker ID in the spec_gen_model\n",
+    "spec_gen_model.fastpitch.speaker_emb.weight.data[new_speaker] = new_speaker_emb"
```

Comment:
Combining this comment with your last comment - "If someone wants to release a model..."

Well, we can do that but I don't think it is super important for this tutorial. Also, that would allow users to use external speaker embeddings which is part of another project (using speaker embeddings from Speaker verification model to condition FastPitch) - I guess this feature is still cooking and will be released with that project.
In speaker interpolation, we do not use speaker embeddings learnt separately. That does not give good results as we have seen with other experiments. Hence I feel it is best we do not separate the speaker embeddings from the multi-speaker model. Separating the speaker embeddings from the model would also remove FastPitch's capability to do multi-speaker inference. Even for normal multi-speaker inference it would be necessary to load both FastPitch and speaker embeddings separately -  which IMO would not be a good experience.

To erase all original speakers we just have to replace their embeddings with interpolated speaker embeddings. I will add a sentence describing this process.

Reply:
For clarification, I was not suggesting anything which would change/remove any of the existing functionality. Just extra optional functions (1) export the embedding weights to a file and (2) import the embedding weights from a file. But to avoid confusion, I agree that these functions and modifying infer() would make more sense after "using speaker embeddings from Speaker verification model to condition FastPitch".

It sounds like your suggestion is to have the user directly assign "spec_gen_model.fastpitch.speaker_emb.weight.data" to a new array, which would be a reasonable workaround for the tutorial.